File: app/Sources/ICD360SVPN/Models/Health.swift
// ICD360SVPN — Models/Health.swift
// MARK: - Health wire model
//
// Mirrors the JSON shape returned by `GET /v1/health`.

import Foundation

/// Liveness + dependency-status snapshot from the agent.
public struct Health: Codable {
    /// `"ok"` if every dependency is up, `"degraded"` otherwise.
    public let status: String

    /// True if `wg show wg0` succeeded.
    public let wgUp: Bool

    /// True if AdGuard Home's `/control/status` returned 200.
    public let adguardUp: Bool

    /// Seconds since the agent process started.
    public let uptimeSeconds: Int64

    /// Build version string of the agent (e.g. `"0.0.1-m2-fix2"`).
    public let agentVersion: String

    /// Server's wall-clock time at the moment the response was built.
    public let serverTime: Date

    enum CodingKeys: String, CodingKey {
        case status
        case wgUp          = "wg_up"
        case adguardUp     = "adguard_up"
        case uptimeSeconds = "uptime_seconds"
        case agentVersion  = "agent_version"
        case serverTime    = "server_time"
    }
}

File: app/lib/src/models/health.dart
// ICD360SVPN — lib/src/models/health.dart

import 'package:flutter/foundation.dart';

@immutable
class Health {
  const Health({
    required this.status,
    required this.wgUp,
    required this.adguardUp,
    required this.uptimeSeconds,
    required this.agentVersion,
    required this.serverTime,
  });

  final String status;
  final bool wgUp;
  final bool adguardUp;
  final int uptimeSeconds;
  final String agentVersion;
  final DateTime serverTime;

  bool get isOk => status == 'ok';

  factory Health.fromJson(Map<String, dynamic> json) {
    return Health(
      status: json['status'] as String,
      wgUp: json['wg_up'] as bool,
      adguardUp: json['adguard_up'] as bool,
      uptimeSeconds: (json['uptime_seconds'] as num).toInt(),
      agentVersion: json['agent_version'] as String,
      serverTime: DateTime.parse(json['server_time'] as String),
    );
  }
}
